### ISSUE TITLE
FindEach cop should not be active with Mongoid

### ISSUE THREAD

**Issue by BorjaL** [rubocop/rubocop#5850]
There is a dependency with ActiveRecord. Class `RuboCop::Cop::FindEach` should be aware not to be active when Mongoid is the ORM, because the method `find_each` does not exist in class `Mongoid::Criteria`

--------

## Expected behaviour

When Mongoid is setting up as ORM, and the method `each` is used in a `Mongoid::Criteria` object, `FindEach` cop should not be triggered because method `find_each` does not exist in `Mongoid::Critera`.

## Actual behavior

If you are using Mongoid and you have code like this:

```
User.where(name: 'John Doe').each do |user|
  # any code
end
```
FindEach cop is active and it gives the following message:

`Rails/FindEach: Use find_each instead of each.`

## Steps to reproduce the problem

Use Mongoid as ORM and use the method `each` in a `Mongoid::Criteria`, for example:
```
User.where(name: 'John Doe').each do |user|
  # any code
end
```

## RuboCop version

```
$ rubocop -V
0.54.0 (using Parser 2.5.0.5, running on ruby 2.3.0 x86_64-darwin16)
```

**Comment by mikegee** [comment]
Rubocop can't know which ORM is behind this code:

```ruby
User.where(name: 'John Doe').each do |user|
  # any code
end
``` 

I suppose this cop could inspect the `Gemfile` to see which ORM is in use. What if both Mongoid and ActiveRecord are found? 🤔

**Comment by bbatsov** [comment]
I think in such cases users can simply disable it, as I'm not sure it's worth making the code more complex for something that basic.

**Comment by Drenmi** [comment]
Once the Rails cops are extracted into their own gem, I guess an `ActiveRecord` department might make sense, since Rails is technically not a single library.
